### PR TITLE
Patient search component

### PR DESCRIPTION
**Pull request by luthfib**
Add patient search component.

image

### CODE REVIEW

**Review by eric-gunther — APPROVED** [review]
Looks good to me, in the interest of urgency for this issue will approve now, but I have 3 follow-up thoughts:

1. We still need to remove the unnecessary `_tag` parameter in searches, but perhaps you have a separate branch for that.
2. Should include in storybook.
3. I think we should show search results with DOB and any non-Zus identifiers, and we should support search by identifier (ie. just updating getBuildersPatientList to search with both the `name` and `identifier` search parameters). I think this is something UX will want to weigh in on so going to put this in Slack.

File: src/components/core/list-box/list-box.tsx [hunk 1]
```diff
@@ -52,58 +54,70 @@ export function ListBox<T extends MinListBoxItem>({
...
+          <>
+            <Listbox.Button
+              className={cx(
+                btnClassName,
+                !useBasicStyles && "after:ctw-bg-content-black",
+                !useBasicStyles &&
+                  "focus-visible:ctw-outline-primary-dark focus-visible:after:ctw-bg-transparent",
+                "ctw-relative ctw-mr-2 ctw-cursor-pointer ctw-border-0 ctw-border-transparent"
+              )}
+            >
```
Comment (mrosata) [inline src/components/core/list-box/list-box.tsx:67]: Does our abstraction of ListBox not work here style-wise? If the styles are diff and it doesn't, that's fine, but if it's the same styles already as the abstraction we should just use`src/components/core/list-box/list-box.tsx`

Comment (mrosata) [reply src/components/core/list-box/list-box.tsx:67]: otherwise pr looks good

**Review by pmadria3 — APPROVED** [review]
LGTM

**Comment by luthfib** [comment]
> Looks good to me, in the interest of urgency for this issue will approve now, but I have 3 follow-up thoughts:
> 
> 1. We still need to remove the unnecessary `_tag` parameter in searches, but perhaps you have a separate branch for that.
> 2. Should include in storybook.
> 3. I think we should show search results with DOB and any non-Zus identifiers, and we should support search by identifier (ie. just updating getBuildersPatientList to search with both the `name` and `identifier` search parameters). I think this is something UX will want to weigh in on so going to put this in Slack.

1. Yes I have separate pr for that.
2. Agreed. 
3. My idea was to get this in and quickly follow-up with a pr to add that. Only reason being I have to add a prop to add a custom render to the each dropdown field.

**Review by eric-gunther — APPROVED** [review]

**Pull request merged**